californium-core/src/main/java/org/eclipse/californium/core/network/deduplication/CropRotation.java

```diff
@@ -114,14 +113,20 @@ public Exchange findPrevious(KeyMID key, Exchange exchange) {
 		return prev;
 	}
 
+	@Override
+	public boolean replacePrevious(KeyMID key, Exchange previous, Exchange exchange) {
+		int s = second;
+		return maps[s].replace(key, previous, exchange) || maps[s].putIfAbsent(key, exchange) == null;
```

Comment:
Maybe we don't care about CropRotation but I don't get why in findPrevious we write in f then in s and here we only write in s.